### PR TITLE
🐛 Destination snowflake: Create final tables with uppercase naming

### LINKED ISSUE #30010: Revert case-senstivity in Snowflake final tables (and don't do it in Redshift)
Steps:
1. Chang the code to not specify case-sensitivity - remove a lot of quotes.
2. Add code that detects if there is a lower-case table name.
    * Make a new `findExistingUpCassedV2Table()` like `findExistingTable()` which will be used within the soft-reset
    * Soft reset (e.g. blow away the loaded_at values in the raw table) 

Notes: 
* don’t force-uppercase things, implicitly do this by not specifying the case and let the DB handle it
* This migration will leave behind the lower-cased old V2 table... that's OK

Future note:
* We also want to rename the raw tables so they are also case-insensitive, but that's another story because (1) that will slow this down and (2) can't be solved by a soft reset.

### PR DESCRIPTION
**Pull request by edgao**
closes https://github.com/airbytehq/airbyte/issues/30010

The interesting changes are in SnowflakeDestinationHandler / SnowflakeSqlGenerator; everything else is just updating tests. (T+D tests are passing locally, at least)

See also https://github.com/airbytehq/airbyte/pull/30068, which implements the migration for existing tables. I'll merge that PR into this one, then release this PR to master.

todo:
* ~~add migration logic. I'll stack a PR on top of this one.~~ https://github.com/airbytehq/airbyte/pull/30068
* ~~run some manual tests~~
* ~~update tests to have uppercase `_airbyte_*` in the expected records. I'll submit an issue to do that - the columns are uppercased in the destination, just the expected records look weird.~~ https://github.com/airbytehq/airbyte/issues/30112

### CODE REVIEW

File: airbyte-integrations/connectors/destination-snowflake/src/test-integration/java/io/airbyte/integrations/destination/snowflake/SnowflakeTestUtils.java [hunk 1]
```diff
@@ -41,9 +45,9 @@ public static List<JsonNode> dumpFinalTable(final JdbcDatabase database, final S
           AND table_name = ?
         ORDER BY ordinal_position;
         """,
-        unescapeIdentifier(databaseName),
-        unescapeIdentifier(schema),
-        unescapeIdentifier(table)).stream()
+        unescapeIdentifier(databaseName).toUpperCase(),
+        unescapeIdentifier(schema).toUpperCase(),
+        unescapeIdentifier(table).toUpperCase()).stream()
```
Comment (evantahler) [inline airbyte-integrations/connectors/destination-snowflake/src/test-integration/java/io/airbyte/integrations/destination/snowflake/SnowflakeTestUtils.java:50]: This is a test util, so maybe it doens't matter, but I thought we wanted to let the DB evaluate the upper-case-ness of table names

**Review by evantahler — COMMENTED** [review]
There's a lot of `toUpperCase()` in the implementation code.  I thought we wanted to let the DB evaluate these these to UPPERCASE lazily?

File: airbyte-integrations/connectors/destination-snowflake/src/test-integration/java/io/airbyte/integrations/destination/snowflake/SnowflakeTestUtils.java (hunk 1 above)
Comment (edgao) [reply airbyte-integrations/connectors/destination-snowflake/src/test-integration/java/io/airbyte/integrations/destination/snowflake/SnowflakeTestUtils.java:50]: I overruled that decision https://airbytehq-team.slack.com/archives/C03C4AVJWG4/p1693495091006359 (but open to continued discussion!)

Comment (evantahler) [reply airbyte-integrations/connectors/destination-snowflake/src/test-integration/java/io/airbyte/integrations/destination/snowflake/SnowflakeTestUtils.java:50]: Yes! 

> with snowflake uppercasing via not-quoting, we’re losing the ability to use special characters / reserved words in namespace/table/column names. That opens up a whole can of worms around escaping things, which I’d rather not deal with right now.

That makes a lot of sense.  👍

Comment (edgao) [reply airbyte-integrations/connectors/destination-snowflake/src/test-integration/java/io/airbyte/integrations/destination/snowflake/SnowflakeTestUtils.java:50]: this is still technically a user-facing decision (e.g. would they rather get `"FOO$BAR"`  or `FOO_BAR`). But that's a much smaller blast radius

**Review by jbfbell — APPROVED** [review]

File: docs/release_notes/upgrading_to_destinations_v2.md [hunk 2]
```diff
@@ -175,12 +175,6 @@ In addition to the changes which apply for all destinations described above, the
 1. [Object and array properties](https://docs.airbyte.com/understanding-airbyte/supported-data-types/#the-types) are properly stored as JSON columns. Previously, we had used TEXT, which made querying sub-properties more difficult.
    - In certain cases, numbers within sub-properties with long decimal values will need to be converted to float representations due to a _quirk_ of Bigquery. Learn more [here](https://github.com/airbytehq/airbyte/issues/29594).
 
-### Snowflake
-
-1. `destination-snowflake` is now case-sensitive, and was not previously. This means that if you have a source stream "users", `destination-snowflake` would have previously created a "USERS" table in your data warehouse. We now correctly create a "users" table.
```
Comment (edgao) [inline docs/release_notes/upgrading_to_destinations_v2.md:180]: @Hesperide fyi - I'm deleting the case-sensitivity thing from snowflake's upgrade guide.

**Comment by edgao** [comment]
source-hubplanner is an unrelated autoformat change. Will approve-and-merge this PR.

**Comment by octavia-approvington** [comment]
Send it! <br/> rocket socks

**Review by octavia-approvington — APPROVED** [review]

**Pull request merged**